packages/arb-token-bridge-ui/src/components/common/SafeImage.tsx

```diff
@@ -11,21 +11,23 @@ export function SafeImage(props: SafeImageProps) {
   const [validImageSrc, setValidImageSrc] = useState<false | string>(false)
 
   useEffect(() => {
+    let isMounted = true
     const image = new Image()
 
     if (typeof src === 'undefined') {
       setValidImageSrc(false)
     } else {
       const sanitizedImageSrc = sanitizeImageSrc(src)
 
-      image.onerror = () => setValidImageSrc(false)
-      image.onload = () => setValidImageSrc(sanitizedImageSrc)
-      image.src = sanitizedImageSrc
+      if (isMounted) {
+        image.onerror = () => setValidImageSrc(false)
+        image.onload = () => setValidImageSrc(sanitizedImageSrc)
+        image.src = sanitizedImageSrc
+      }
     }
 
-    return function cleanup() {
-      // Abort previous loading
-      image.src = ''
+    return () => {
+      isMounted = false
```

Comment:
We still need to cleanup the `src` here to abort the download

Reply:
+1 need to add back the original cleanup code

Reply:
hmm ok let me think about it because it throws an error if updates on an unmounted component

Reply:
I brought this one back as it was, it should be working properly but still throwing an error - I think it's a bit of a misleading error and will be gone once we migrate to Next